### PR TITLE
Fix: SQL ecaping and bind naming

### PR DESCRIPTION
**Pull request by Meldiron**
We already do:

- Wrap custom keys in ` symbols to prevent sql syntax errors with dash symbols in attributes
- Use number instead of key in bind name, same reason as above

So for instance, with attribute name `e-mail`, ideally we would try to do following:

```
WHERE tables_1.`e-mail` = :attribute_0
```

Notice attribute key is inside `, and bind name does not have the name, instead, a number. This prevents all errors with running the statement later.

---

Problem is, this was not consistent - it was not applied in all places. This PR applies it everywhere.

- [x] New tests added

Blocked by this PR: https://github.com/utopia-php/database/pull/175

> Mongo is failing, but this should be resolved by the PR linked above

### CODE REVIEW

**Comment by Meldiron** [comment]
Mongo test is failing with new tests, looks like we have the same problem with dashes there.

**Comment by stnguyen90** [comment]
Related Issue: 

* https://github.com/appwrite/appwrite/issues/3600#issuecomment-1271752007?

**Comment by stnguyen90** [comment]
Hmm...I'm having trouble testing this because I'm getting this error on startup:

```
appwrite  | [Setup] - Server database init started...
appwrite  | [Setup] - Creating metadata table: appwrite...
appwrite  | [Setup] - Skip: metadata table already exists
appwrite  | 
appwrite  | Fatal error: Uncaught PDOException: Table 'appwrite._console_metadata' doesn't exist in @swoole-src/library/core/Database/PDOStatementProxy.php:64
appwrite  | Stack trace:
appwrite  | #0 /usr/src/code/vendor/utopia-php/database/src/Database/Adapter/MariaDB.php(429): Swoole\Database\PDOStatementProxy->__call('execute', Array)
appwrite  | #1 /usr/src/code/vendor/utopia-php/database/src/Database/Database.php(1182): Utopia\Database\Adapter\MariaDB->getDocument('metadata', 'audit')
appwrite  | #2 /usr/src/code/vendor/utopia-php/database/src/Database/Database.php(468): Utopia\Database\Database->getDocument(Object(Utopia\Database\Document), 'audit')
appwrite  | #3 /usr/src/code/app/http.php(109): Utopia\Database\Database->getCollection('audit')
appwrite  | #4 {main}
appwrite  |   thrown in @swoole-src/library/core/Database/PDOStatementProxy.php on line 64
appwrite  | [2022-10-18 18:56:49 #1.8]  ERROR   php_swoole_server_rshutdown() (ERRNO 503): Fatal error: Uncaught PDOException: Table 'appwrite._console_metadata' doesn't exist in @swoole-src/library/core/Database/PDOStatementProxy.php:64
appwrite  | Stack trace:
appwrite  | #0 /usr/src/code/vendor/utopia-php/database/src/Database/Adapter/MariaDB.php(429): Swoole\Database\PDOStatementProxy->__call('execute', Array)
appwrite  | #1 /usr/src/code/vendor/utopia-php/database/src/Database/Database.php(1182): Utopia\Database\Adapter\MariaDB->getDocument('metadata', 'audit')
appwrite  | #2 /usr/src/code/vendor/utopia-php/database/src/Database/Database.php(468): Utopia\Database\Database->getDocument(Object(Utopia\Database\Document), 'audit')
appwrite  | #3 /usr/src/code/app/http.php(109): Utopia\Database\Database->getCollection('audit')
appwrite  | #4 {main}
appwrite  |   thrown in @swoole-src/library/core/Database/PDOStatementProxy.php on line 64
```

**Review by stnguyen90 — COMMENTED** [review]
Hmm...I'm having trouble testing this because I'm getting this error on startup:

```
appwrite  | [Setup] - Server database init started...
appwrite  | [Setup] - Creating metadata table: appwrite...
appwrite  | [Setup] - Skip: metadata table already exists
appwrite  | 
appwrite  | Fatal error: Uncaught PDOException: Table 'appwrite._console_metadata' doesn't exist in @swoole-src/library/core/Database/PDOStatementProxy.php:64
appwrite  | Stack trace:
appwrite  | #0 /usr/src/code/vendor/utopia-php/database/src/Database/Adapter/MariaDB.php(429): Swoole\Database\PDOStatementProxy->__call('execute', Array)
appwrite  | #1 /usr/src/code/vendor/utopia-php/database/src/Database/Database.php(1182): Utopia\Database\Adapter\MariaDB->getDocument('metadata', 'audit')
appwrite  | #2 /usr/src/code/vendor/utopia-php/database/src/Database/Database.php(468): Utopia\Database\Database->getDocument(Object(Utopia\Database\Document), 'audit')
appwrite  | #3 /usr/src/code/app/http.php(109): Utopia\Database\Database->getCollection('audit')
appwrite  | #4 {main}
appwrite  |   thrown in @swoole-src/library/core/Database/PDOStatementProxy.php on line 64
appwrite  | [2022-10-18 18:56:49 #1.8]  ERROR   php_swoole_server_rshutdown() (ERRNO 503): Fatal error: Uncaught PDOException: Table 'appwrite._console_metadata' doesn't exist in @swoole-src/library/core/Database/PDOStatementProxy.php:64
appwrite  | Stack trace:
appwrite  | #0 /usr/src/code/vendor/utopia-php/database/src/Database/Adapter/MariaDB.php(429): Swoole\Database\PDOStatementProxy->__call('execute', Array)
appwrite  | #1 /usr/src/code/vendor/utopia-php/database/src/Database/Database.php(1182): Utopia\Database\Adapter\MariaDB->getDocument('metadata', 'audit')
appwrite  | #2 /usr/src/code/vendor/utopia-php/database/src/Database/Database.php(468): Utopia\Database\Database->getDocument(Object(Utopia\Database\Document), 'audit')
appwrite  | #3 /usr/src/code/app/http.php(109): Utopia\Database\Database->getCollection('audit')
appwrite  | #4 {main}
appwrite  |   thrown in @swoole-src/library/core/Database/PDOStatementProxy.php on line 64
```

**Comment by abnegate** [comment]
@Meldiron Please fix the conflicts here and get the tests running, the blocking PR has been merged

**Comment by Meldiron** [comment]
@abnegate updated, tests passing locally.

File: src/Database/Adapter/MariaDB.php [hunk 1]
```diff
@@ -878,8 +878,11 @@ public function find(string $collection, array $queries = [], int $limit = 25, i
             });
 
             $conditions = [];
+            $attributeIndex = 0;
             foreach ($query->getValues() as $key => $value) {
-                $conditions[] = $this->getSQLCondition('table_main.`' . $query->getAttribute().'`', $query->getMethod(), ':attribute_' . $i . '_' . $key . '_' . $query->getAttribute(), $value);
+                $bindKey = 'key_' . $attributeIndex;
+                $conditions[] = $this->getSQLCondition('table_main.`' . $query->getAttribute() . '`', $query->getMethod(), ':attribute_' . $i . '_' . $key . '_' . $bindKey, $value);
+                $attributeIndex++;
```
Comment (abnegate) [inline src/Database/Adapter/MariaDB.php:885]: We can use a post-increment here to make the intention clear on the same line as the usage:

```suggestion
                $bindKey = 'key_' . $attributeIndex++;
                $conditions[] = $this->getSQLCondition('table_main.`' . $query->getAttribute() . '`', $query->getMethod(), ':attribute_' . $i . '_' . $key . '_' . $bindKey, $value);
```

Applies to everywhere we're using `$attributeIndex` in this PR

Comment (abnegate) [reply src/Database/Adapter/MariaDB.php:885]: Actually I see now the way you've done it is consistent with the rest of the codebase, resolving this comment :+1:

**Review by abnegate — APPROVED** [review]

**Pull request merged**